D/D.sublime-syntax

```diff
@@ -8,339 +8,1971 @@ file_extensions:
+      set: [value-parens-after, value-list]
+    - match: (?=\S)
+      set: value-after
+  new-anonymous-after-args:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      set:
+        - match: '(?={)'
+          set: optional-template-body-block
+        - match: '(?=\S)'
+          set: [optional-template-body-block, base-class-list]
+    - include: not-whitespace-illegal
+  is-after-type:
+    - match: '\b({{name}})\b'
+      scope: variable.parameter.d
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - match: '\=\=|:'
+      scope: keyword.operator.logical.d
+      set: [is-after, type-specialization]
+  is-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: function-argument-definition-list-content
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - include: not-whitespace-illegal
+  type-specialization:
+    - match: '\b(?:({{type_specialization_storage}})|({{type_specialization_keyword}}))\b'
+      captures:
+        1: storage.type.d
+        2: keyword.d
+      set:
+        - match: '(?=\))'
+          pop: true
+        - include: type-specialization
+    - match: '(?=\S)'
+      set: type
+    - include: not-whitespace-illegal-pop
+  value-group-after:
+    - match: '\b(function|delegate)\b'
+      scope: keyword.other.d
+      set: [value-group-after, basic-type2, function-type]
+    - match: '\)'
+      scope: punctuation.section.group.end.d
+      set: value-group-after-parens
+    - match: '\b({{name}})\b'
+      scope: variable.parameter.d
+      set: value-group-after
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set: [value-group-after-parens, function-argument, function-argument-or-type]
+    - match: '\.\.\.'
+      scope: keyword.operator.d
+    - include: not-whitespace-illegal
+  value-group-after-parens:
+    - include: function-attribute-in
+    - match: (?={)
+      set: [value-after, block-statement]
+    - match: '=>'
+      scope: keyword.operator.lambda.d
+      set: value
+    - match: (?=\S)
+      set: value-after
+  value-after:
+    - match: '\.\.(?!\.)'
+      scope: keyword.operator.d
+      set: value
+    - match: '\.(?!\.\.)'
+      scope: punctuation.accessor.dot.d
+      set:
+        - match: '{{identifier}}'
+          scope: variable.other.d
+          set: value-after
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set: [value-parens-after, value-list]
+    - match: '\['
+      scope: punctuation.section.brackets.begin.d
+      set: [value-bracket-after, value-list]
+    - match: '(\+\+|--)'
+      scope: keyword.operator.arithmetic.d
+      set: value-after
+    - match: '(&&|\|\|)'
+      scope: keyword.operator.logical.d
+      set: value
+    - match: '(\+|-|/|%|\*|&|\||\^\^|\^|~|<<|>>>|>>)='
+      scope: keyword.operator.assignment.d
+      set: value
+    - match: '(\+|\-|~|/|%|\^\^)'
```

Comment:
matching + and - in two rules will cause the second one to be ignored. Should be matched once, only.

Reply:
The first rule ends with a `=` though, so the 2nd one would trigger if there isn't a `=` after.

Reply:
Sorry, my fault.